tests/test_backend.py

```diff
@@ -506,7 +515,7 @@ def test_reshape_dynamic(self):
         x_val = np.array([1.0, 2.0, 3.0, 4.0], dtype=np.float32).reshape((2, 2))
         x = tf.placeholder(tf.float32, [2, 2], name=_TFINPUT)
         shape = tf.constant([4, 1])
-        x_ = tf.reshape(x, tf.transpose(shape))
+        x_ = tf.reshape(x, shape)
```

Comment:
may I know the reason for this? 
If we need do this,  test_reshape_dynamic should be redundant , test_reshape already cover the same case maybe?

Reply:
turns out onnxruntime transpose works only on float. In the past that worked only because the onnx optimizer removed this transpose because perm=[0].
But now that I disabled the onnx optimizer the transpose on int is there and onnxruntime ran into it.
And of course that unit test never did what we wanted it to do since the transpose was removed.
You are right, I need to do something with the shape input - maybe pass it as placeholder will make it really dynamic,

Reply:
get it! thanks! yeah, making it into placeholder should work.

Reply:
ok, using placeholder now